planet/src/main/FirebaseSecurityRules.txt

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// This file contains the security rules of the firebase realtime databse
+{
+    "rules": {
+        // Rules applied to the users tree
+        // Users can read other users' data and write their own data
+        "users": {
+            "$uid": {
+                ".read": true,
+                ".write": "$uid === auth.uid"
+            }
+        },
+        
+        // Rules applied to the places tree
+        // Users can read all places data, but can only edit their own data
+        "places": {
+          	".read": true,
+            "$placeId":{
+                "visitors":{
+                    "$uid" : {
+                        ".write": "$uid === auth.uid"
+                    }
+                }
+            }
+        },
+        
+        // Rules applied to the events tree
+        // Users can read other users' data and write their own data
+        "events": {
+            "$uid": {
+                ".read": true,
+                ".write": "$uid === auth.uid",
+                "$listName": {
+                    ".indexOn": ["order"]
```

Comment:
why do we need the index on order?

Reply:
Firebase provides indexing such that it can improve query performance when we do sorting/filtering. 
This website talks a little about it: https://howtofirebase.com/firebase-security-rules-88d94606ce4a 
![image](https://user-images.githubusercontent.com/46226072/88341882-666de080-cd0c-11ea-83a6-a6018cfa1aa9.png)

Reply:
in what cases do we search for an even with a specific order?

Reply:
Every time when we're rendering a list of events, we need to order them by the "order" property. (The order property gets changed every time users reorder the cards on the font end)

Reply:
An index will make a query for the entry with a specific value of the indexed child very fast. It will not change the results at all if you're fetching all of the items.

It sounds from your description like you're loading all events anyways, and then sorting them. An index won't help with that.

Reply:
Ahh okayy, I'll remove it then